src/js/control-bar/text-track-controls/chapters-button.js

```diff
@@ -39,107 +40,90 @@ class ChaptersButton extends TextTrackButton {
     return `vjs-chapters-button ${super.buildCSSClass()}`;
   }
 
-  /**
-   * Create a menu item for each text track
-   *
-   * @return {Array} Array of menu items
-   * @method createItems
-   */
-  createItems() {
-    let items = [];
+  update(event) {
+    if (this.track_ === undefined || (event && (event.type === 'addtrack' || event.type === 'removetrack'))) {
+      this.setTrack(this.findChaptersTrack());
+    }
+    super.update();
+  }
+  
+  setTrack(track) {
+    if (this.track_ === track) return;
+
+    if (this.track_) {
+      let remoteTextTrackEl = this.player_.remoteTextTrackEls().getTrackElementByTrack_(this.track_);
+      if (remoteTextTrackEl) {
+        remoteTextTrackEl.removeEventListener('load', this.updateHandler_);
+      }
+      this.track_ = null;
+    }
+
+    this.track_ = track;
 
-    let tracks = this.player_.textTracks();
+    if (this.track_) {
+      this.track_['mode'] = 'hidden';
 
-    if (!tracks) {
-      return items;
+      let remoteTextTrackEl = this.player_.remoteTextTrackEls().getTrackElementByTrack_(this.track_);
+      if (remoteTextTrackEl) {
+        remoteTextTrackEl.addEventListener('load', this.updateHandler_);
+      }
     }
+  }
 
-    for (let i = 0; i < tracks.length; i++) {
+  findChaptersTrack() {
+    let tracks = this.player_.textTracks() || [];
+    
+    for (let i = tracks.length - 1; i >= 0; i--) {
+      // We will always choose the last track as our chaptersTrack
       let track = tracks[i];
+
       if (track['kind'] === this.kind_) {
-        items.push(new TextTrackMenuItem(this.player_, {
-          'track': track
-        }));
+        return track;
       }
     }
+  }
 
-    return items;
+  getMenuCaption() {
+    if (this.track_ && this.track_.label)
+      return this.track_.label;
+    return this.localize(toTitleCase(this.kind_));
   }
 
   /**
-   * Create menu from chapter buttons
+   * Create menu from chapter track
    *
    * @return {Menu} Menu of chapter buttons
    * @method createMenu
    */
   createMenu() {
-    let tracks = this.player_.textTracks() || [];
-    let chaptersTrack;
-    let items = this.items || [];
-
-    for (let i = tracks.length - 1; i >= 0; i--) {
+    this.options_.title = this.getMenuCaption();
+    return super.createMenu();
+  }
 
-      // We will always choose the last track as our chaptersTrack
-      let track = tracks[i];
+  /**
+   * Create a menu item for each chapter cue
+   *
+   * @return {Array} Array of menu items
+   * @method createItems
+   */
+  createItems() {
```

Comment:
Yes, it was too big rewrite so diffs don't make too much sense here. Thank you, have a nice weekend.